cmd/snap-bootstrap/initramfs_systemd_mount.go

```diff
@@ -149,6 +174,37 @@ func doSystemdMountImpl(what, where string, opts *systemdMountOptions) error {
 	if opts.Private {
 		options = append(options, "private")
 	}
+	if opts.Overlayfs {
+		if len(opts.LowerDirs) <= 0 || len(opts.UpperDir) <= 0 || len(opts.WorkDir) <= 0 {
+			return fmt.Errorf("cannot mount %q at %q: missing arguments for overlayfs mount. lowerdir, upperdir, workdir are needed.", what, where)
+		}
+
+		if strings.ContainsAny(opts.UpperDir, forbiddenChars) {
+			return fmt.Errorf("cannot mount %q at %q: upperdir overlayfs mount option contains forbidden characters. %q contains one of \"%s\".", what, where, opts.UpperDir, forbiddenChars)
+		}
+		if strings.ContainsAny(opts.WorkDir, forbiddenChars) {
+			return fmt.Errorf("cannot mount %q at %q: workdir overlayfs mount option contains forbidden characters. %q contains one of \"%s\".", what, where, opts.WorkDir, forbiddenChars)
+		}
+
+		var lowerDirs strings.Builder
+		for i, d := range opts.LowerDirs {
+			if i != 0 {
+				lowerDirs.WriteRune(':')
+			}
+
+			if strings.ContainsAny(d, forbiddenCharsForOverlayfsLowerDir) {
+				return fmt.Errorf("cannot mount %q at %q: lowerdir overlayfs mount option contains forbidden characters. %q contains one of \"%s\".", what, where, d, forbiddenCharsForOverlayfsLowerDir)
+			}
+
+			// This is used for splitting multiple lowerdirs as done in
+			// https://elixir.bootlin.com/linux/v6.10.9/C/ident/ovl_parse_param_split_lowerdirs
+			ed := strings.ReplaceAll(d, ":", `\:`)
```

Comment:
Do not we forbid this character?

Reply:
I see we do not. We do it only for upper and work dir. But that makes little sense.

Reply:
We don't for lowerdirs. lowerdirs are using the `forbiddenCharsExceptColon` variable which now was renamed to `forbiddenCharsForOverlayfsLowerDir`